### PR TITLE
MyRocks clone: add missing donor mutex unlocks on error exits

### PR DESCRIPTION
**Pull request by laurynas-biveinis**

### CODE REVIEW

**Comment by laurynas-biveinis** [comment]
Scope guards should be used instead, but that's outside the scope of this fix.

File: storage/rocksdb/clone/donor.cc [hunk 1]
```diff
@@ -917,17 +917,26 @@ int donor::copy(const THD *thd, uint task_id, Ha_clone_cbk &cbk) {
...
+      return err;
+    }
 
     if (should_roll_checkpoint()) {
       std::size_t checkpoint_size_delta;
       err = next_checkpoint_locked(false, checkpoint_size_delta);
       mysql_mutex_unlock(&m_donor_mutex);
-      if (err != 0) return err;
+      if (err != 0) {
+        mysql_mutex_unlock(&m_donor_mutex);
```
Comment (hermanlee) [inline storage/rocksdb/clone/donor.cc]: line 928 unlocks the mutex.

File: storage/rocksdb/clone/donor.cc [hunk 2]
```diff
@@ -917,17 +917,26 @@ int donor::copy(const THD *thd, uint task_id, Ha_clone_cbk &cbk) {
...
+        return err;
+      }
 
       const donor_estimate clone_size_estimate_delta{checkpoint_size_delta};
       err = send_metadata(clone_size_estimate_delta, cbk);
-      if (err != 0) return err;
+      if (err != 0) {
+        mysql_mutex_unlock(&m_donor_mutex);
```
Comment (hermanlee) [inline storage/rocksdb/clone/donor.cc]: line 928 unlocks the mutex.

File: storage/rocksdb/clone/donor.cc (hunk 1 above)
Comment (laurynas-biveinis) [reply storage/rocksdb/clone/donor.cc]: Oh. Fixed.

**Comment by laurynas-biveinis** [comment]
Ready for review

**Pull request closed without merging**